py/ai/h2o/sparkling/ml/features/H2OTargetEncoder.py

```diff
@@ -0,0 +1,107 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from pyspark.ml.util import JavaMLWritable
+from pyspark.ml.wrapper import JavaEstimator
+from pyspark.sql import SparkSession
+from pyspark import keyword_only
+from py_sparkling.ml.util import getValidatedEnumValue, validateEnumValue
+
+from pysparkling.context import H2OContext
+from pysparkling.spark_specifics import get_input_kwargs
+from ai.h2o.sparkling.ml.params import H2OTargetEncoderParams
+from py_sparkling.ml.util import set_double_values, JavaH2OMLReadable
+from ai.h2o.sparkling.ml.models import H2OTargetEncoderModel
+
+from h2o.utils.typechecks import assert_is_type, Enum
+
+
+class H2OTargetEncoder(H2OTargetEncoderParams, JavaEstimator, JavaH2OMLReadable, JavaMLWritable):
+
+    @keyword_only
+    def __init__(self, foldCol=None, labelCol="label", inputCols=[], holdoutStrategy = "None",
+                 blendedAvgEnabled=False, blendedAvgInflectionPoint=10.0, blendedAvgSmoothing=20.0, noise=0.01, noiseSeed=-1):
+        super(H2OTargetEncoder, self).__init__()
+        self._hc = H2OContext.getOrCreate(SparkSession.builder.getOrCreate(), verbose=False)
+        self._java_obj = self._new_java_obj("ai.h2o.sparkling.ml.features.H2OTargetEncoder", self.uid)
```

Comment:
Just FYI: When working on kmeans, I had a problem with putting just kmeans into a single file. It occured during loading the serialized pipeline because it thought the name of the file itself is part of the java package.

Reply:
I will cover that with tests. Thanks a lot!

Reply:
For now I'm using the shared file

Reply:
Do you use the new package?

Reply:
Yup. But let's say my file was at `at/h2o/sparking/algos/kmeans.py`

Then I guess the readable tried to read it from scala as `ai.h2o.sparkling.algos.kmeans.H2OKmeans`

I have idea how to fix that using custom JavaMLReadable, but would like to focus on this after release as this change would not affect users

Reply:
Hope my [test](https://github.com/h2oai/sparkling-water/pull/1354/files#diff-084cb88c118d3617b8cbe7299394efadR70) covers the scenario properly. It seems it works for both cases with `JavaH2OMLReadable` and `JavaMLReadable`.

Have you tried to add:
```
from ai.h2o.sparkling.algos.kmeans  import H2OKMeans
```
to `ai/h2o/sparking/algos/__init__.py`?

Reply:
Let's have a all on Monday

Reply:
The import did the trick as the package name is the one Spark expects now. Yup, the test is good!

Do we also have a test where we serialize and de serialize the fitted TE Model?

Reply:
The import did the trick as the package name is the one Spark expects now. Yup, the test is good!

Do we also have a test where we serialize and de serialize the fitted TE Model?

Reply:
The below tests should cover various aspects of serde. Are they ok?
- [a test testing prediction abilities](https://github.com/h2oai/sparkling-water/pull/1354/files#diff-084cb88c118d3617b8cbe7299394efadR119)
- [propagation of parameters to mojo model] - [here](https://github.com/h2oai/sparkling-water/pull/1354/files#diff-084cb88c118d3617b8cbe7299394efadR57) and [here](https://github.com/h2oai/sparkling-water/pull/1354/files#diff-084cb88c118d3617b8cbe7299394efadR88).

Reply:
Perfect, thanks!!